cache/lru/lru.go

```diff
@@ -74,7 +102,9 @@ func (tc *TieredLRUCache) AddObject(oi *types.ObjectIndex) error {
 	tc.mutex.Lock()
 	defer tc.mutex.Unlock()
 
-	if _, ok := tc.lookup[oi.HashStr()]; ok {
+	var oiPath = objectIndexToPath(oi)
+	if ok := tc.lookup.Has(oiPath); ok {
+		//!TODO: Create AlreadyInCacheErr type which implements the error interface
```

Comment:
That TODO is obsolete, it was recently fixed. Probably incorrectly rebased from master.